e2e/tests/datasets/dataset-create.spec.ts

```diff
@@ -102,7 +103,9 @@ describe('Create Dataset', () => {
     const dataSetName = await saveModal.saveDataset();
 
     // Verify create successful.
-    await dataPage.openTab(TabLabelAlias.Datasets);
+    await dataPage.openTab(TabLabelAlias.Datasets, {waitPageChange: false});
+    await waitWhileLoading(page);
```

Comment:
Adding a bunch of these makes me a bit worried about maintenance. Is there a reason these needed to be added here? vs just waiting for the next button to become clickable/undisabled?

Reply:
Admittedly, this is not 'clean code'. 

`waitWhileLoading` function checks for existences of spinners spinning for up to one second. Each time a call to function is made, it adds one extra second in playback time when no spinner was found.

thru observation, app UI performance is unreliable. A conscious decision was made to make tests more resilient to failures related to page's readiness. It is a tradeoff between having longer playback time and "flaky" tests.